### ISSUE TITLE
[F08] wrong results with contiguous pointer to array section

### ISSUE THREAD

**Issue by janusw** [flang-compiler/flang#338]
Please consider this simple test program (valid Fortran 2008 AFAICS):

```
program ptr_to_arr_section

   implicit none
   integer, allocatable, dimension(:), target :: arr
   integer, pointer,     dimension(:), contiguous :: ptr

   allocate(arr(1:2), source = (/ 1, 2 /) )

   print *,arr(2:)   ! ok
   ptr => arr(2:)
   print *, ptr(:)   ! still ok
   print *, ptr      ! wrong !!!

end
```

While gfortran & ifort give the correct answer,
```
            2
            2
            2
```
flang & PGI 17.10 print
```
            2
            2
            0
```
which is obviously wrong. A known workaround is to remove the CONTIGUOUS attribute.

**Comment by janusw** [comment]
It is probably not too surprising that valgrind also shows lots of issues on the line which triggers the wrong-code bug:

```
$ flang ptr_to_arr_section.f90 
$ valgrind ./a.out 
==25263== Memcheck, a memory error detector
==25263== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==25263== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==25263== Command: ./a.out
==25263== 
            2
            2
==25263== Conditional jump or move depends on uninitialised value(s)
==25263==    at 0x4EC1B71: __fortio_fmt_i (in /home/janus/llvm5/install/lib/libflang.so)
==25263==    by 0x4EC18BF: __fortio_default_convert (in /home/janus/llvm5/install/lib/libflang.so)
==25263==    by 0x4EE0614: __f90io_ldw (in /home/janus/llvm5/install/lib/libflang.so)
==25263==    by 0x4ED4497: __io_write (in /home/janus/llvm5/install/lib/libflang.so)
==25263==    by 0x4ED43BD: __fortio_main (in /home/janus/llvm5/install/lib/libflang.so)
==25263==    by 0x4010E0: MAIN_ (ptr_to_arr_section.f90:12)
==25263==    by 0x401145: main (in /home/janus/fort/flang_bugs/338/a.out)
```

**Comment by janusw** [comment]
I'm happy to see that this problem seems to be fixed on today's master, supposedly due to 8e148b419b799870845debbe6a1ce2b57c574379. Thank you :)

**Comment by janusw** [comment]
Btw, wouldn't it be a good idea to add a test for this case to flang's testsuite as well?

From my experiences with gfortran I conclude say that it's hard to overestimate the value of a proper testing coverage. See also the recent discussions at https://groups.google.com/forum/#!topic/comp.lang.fortran/AIHRQ2kJv3c
